op-acceptor/runner/runner.go

```diff
@@ -423,79 +431,167 @@ func isValidTestName(name string) bool {
 	return true
 }
 
-// runTestList runs a list of tests and aggregates their results
+// runTestList runs multiple tests in the same package
 func (r *runner) runTestList(metadata types.ValidatorMetadata, testNames []string) (*types.TestResult, error) {
-	if len(testNames) == 0 {
-		r.log.Warn("No tests found to run in package", "package", metadata.Package)
-		return &types.TestResult{
-			Metadata: metadata,
-			Status:   types.TestStatusSkip,
-			Duration: 0,
-			SubTests: make(map[string]*types.TestResult),
-		}, nil
-	}
-
-	var result types.TestStatus = types.TestStatusPass
-	var testErrors []error
-	var totalDuration time.Duration
-	testResults := make(map[string]*types.TestResult)
-
-	// Run each test in the list
+	baseMetadata := metadata
+	baseMetadata.RunAll = false
+
+	var allErrors []string
+	anySkipped := false
+
+	// Log to structured logger
+	r.log.Info("Running tests in package", "package", metadata.Package, "testCount", len(testNames))
+
+	// Create a combined result for all tests in the package
+	result := &types.TestResult{
+		Metadata: metadata,
+		Status:   types.TestStatusPass,               // Default to pass, will be updated below
+		SubTests: make(map[string]*types.TestResult), // Initialize SubTests map
+	}
+
 	for _, testName := range testNames {
-		// Create a new metadata with the specific test name
-		testMetadata := metadata
+		testMetadata := baseMetadata
 		testMetadata.FuncName = testName
 
-		// Run the individual test
 		testResult, err := r.runSingleTest(testMetadata)
 		if err != nil {
-			return nil, fmt.Errorf("running test %s: %w", testName, err)
+			allErrors = append(allErrors, fmt.Sprintf("Test %s error: %v", testName, err))
+			continue
 		}
 
-		// Store the individual test result
-		testResults[testName] = testResult
-		totalDuration += testResult.Duration
+		// Store this test as a subtest
+		result.SubTests[testName] = testResult
 
-		// Update overall status based on individual test result
-		if testResult.Status == types.TestStatusFail {
-			result = types.TestStatusFail
+		// Check test status
+		switch testResult.Status {
+		case types.TestStatusFail:
 			if testResult.Error != nil {
-				testErrors = append(testErrors, fmt.Errorf("%s: %w", testName, testResult.Error))
+				allErrors = append(allErrors, fmt.Sprintf("Test %s failed: %v", testName, testResult.Error))
+			} else {
+				allErrors = append(allErrors, fmt.Sprintf("Test %s failed with no error message", testName))
 			}
+		case types.TestStatusSkip:
+			anySkipped = true
 		}
 	}
 
-	// Create the aggregate result
-	return &types.TestResult{
-		Metadata: metadata,
-		Status:   result,
-		Error:    errors.Join(testErrors...),
-		Duration: totalDuration,
-		SubTests: testResults,
-	}, nil
+	// Determine overall status
+	if len(allErrors) > 0 {
+		result.Status = types.TestStatusFail
+	} else if anySkipped {
+		result.Status = types.TestStatusSkip
```

Comment:
is that desirable? (I'm genuinely wondering)
Normally skipped tests don't contribute to the overall status. Although I definitely  do understand the desire to keep track of whether there were some skipped tests.
I dunno, it feels potentially confusing

Reply:
Yeah, I see your point.
I may hold off on merging this PR for now, switch to JSON, then try it again.